development/Dockerfile

```diff
@@ -41,7 +41,7 @@ COPY . /source
 RUN cp /tmp/install/* /source/
 RUN poetry install --no-interaction --no-ansi
 
-# TODO: Remove this, once there is a proper local doc hosting strategy
-RUN pip install -r docs/requirements.txt
+# Install build dependencies for docs generation
+RUN poetry add $(cat docs/requirements.txt)
```

Comment:
This seems odd, is this a preferred pattern.

Reply:
The `docs/requirements.txt` is needed by the RTD build process. I don't have a better way of avoiding duplication of pinned deps at the moment.

It can take stuff out of `pyproject.toml` according to their docs but I don't think you want the mkdocs dependencies in the main project deps?

Reply:
The other option is just explicitly adding the requirements to `pyproject.toml` long with `docs/requirements.txt`.

Reply:
> The other option is just explicitly adding the requirements to `pyproject.toml` along with `docs/requirements.txt`. Reminder to also always use explicitly pinned (`==`) versions for docs dependencies!

This is what core does specifically for this exact reason: a narrow place where duplication does weigh out in complexity.